src/runner/redis.ts

```diff
@@ -147,15 +137,25 @@ class RedisRunner extends BaseRunner implements IRunner {
   }
 
   async process(topics?: string[]) {
-    const loop = () =>
-      setTimeout(
+    const loop = () => {
+      if (this.steveo.exiting) return;
+      if (this.currentTimeout) clearTimeout(this.currentTimeout);
+
+      this.currentTimeout = setTimeout(
         this.process.bind(this, topics),
-        (this.config as RedisConfiguration).consumerPollInterval ?? 1000
+        this.config.consumerPollInterval ?? 1000
       );
-    await this.checks(loop);
+    };
+
+    if (this.paused) {
+      this.logger.debug(`paused processing`);
+      return;
```

Comment:
Should this call `loop()` ? Might just be the github formatting confusing me but it looks like it will break out of process and not actually be triggered again to check for unpausing